lib/toy_module.rb

```diff
@@ -8,6 +8,7 @@ def self.toy_new
   def initialize
     @constant_map = {}
     @method_map = {}
+    super(self.class)
```

Comment:
This works, but the more regular approach is to match what happens in the superclass, `ToyObject`: the constructor takes a “what class am I an instance of?” argument, and `self.toy_new` passes `self` into that constructor.

If we do the same thing here, we can a) avoid calling `self.class` inside the constructor like this (because the constructing class has already supplied its own identity), and b) have a `self.toy_new` definition that’s identical to the superclass’s, so we can actually inherit it rather than reimplement it if we want to. I think it’ll also be easier for us to keep track of what each constructor does and where all the information originates.